src/mgmt/mgmt_common.c

```diff
@@ -132,9 +133,22 @@ ds_mgmt_tgt_recreate(uuid_t pool_uuid, daos_size_t scm_size, int tgt_nr, daos_si
 		goto out;
 	}
 
+	D_ASSERT(tgt_nr > 0);
```

Comment:
[Minor]We could move check if (bio_nvme_configured(SMD_DEV_TYPE_META)) here and only allocate and set bitmap for MD-on-SSD case here.